### PR TITLE
[FIF-271] Adds validation to the installer so the user is forced to provide goo…

### PR DESCRIPTION
**Pull request by DavidJGapCR**
…gle or adfs authentication configuration values.

### Testing
Before you execute \EdFi.Buzz.Installer\eng\Windows\install.ps1, change the Google and ADFS values. For example: 
1. Add the googleClientId but keep the clientSecret null or empty. Or the other way around.
2. Change the idProvider value to adfs and then add a value for adfsClientId, but keep the adfsTenantId null, or the other way around. 
3. Etc.
After all the failure scenarios you can think of are tested. Make sure you test a success scenario with Google and then with ADFS.

### CODE REVIEW

File: EdFi.Buzz.Installer/eng/Windows/Application/appinstallValidations.psm1 [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+#Requires -version 5
+#Requires -RunAsAdministrator
+
+function Authentication-Configuration-Valid {
```
Comment (kurtjohnson-HS) [inline EdFi.Buzz.Installer/eng/Windows/Application/appinstallValidations.psm1:9]: PowerShell is complaining about the verb name, and the extra hyphen. Usually PowerShell functions have a Verb-Phrase format - Install-BuzzApp. And the verb should be approved.

I would recommend just globally changing Authentication-Configuration-Valid to Test-AuthConfiguration everywhere. That should get around it...

You can check the PS verbs with this command....
`Get-Verb | ? { $_.Verb -like '*test*'}`

**Review by kurtjohnson-HS — CHANGES_REQUESTED** [review]
We should change the configuration checking function to make PowerShell not complain....

image

File: EdFi.Buzz.Installer/eng/Windows/Application/appinstallValidations.psm1 (hunk 1 above)
Comment (kurtjohnson-HS) [reply EdFi.Buzz.Installer/eng/Windows/Application/appinstallValidations.psm1:9]: Also, appinstall.ps1 has `Check-BuzzServices`. Check is not an approved verb, 

We should also change that to `Test-BuzzServices`.

Comment (DavidJGapCR) [reply EdFi.Buzz.Installer/eng/Windows/Application/appinstallValidations.psm1:9]: Right... I saw the warning and I thought: I'll deal with that later. And then I forgot.  Thanks!

Comment (DavidJGapCR) [reply EdFi.Buzz.Installer/eng/Windows/Application/appinstallValidations.psm1]: Fixed.

**Pull request merged**